### ISSUE TITLE
View Message Edit History

### ISSUE THREAD

**Issue by tekook** [RocketChat/Rocket.Chat#11370]
I don't know if this is already implemented and I only don't how to access this.
Under Settings -> Messages there is a setting to keep the message history "Message_KeepHistory".
Will this setting keep the real changes / original messages, or does it just keep track of who edited the message?

We are using RocketChat in a corparate enviroment and thus we need to keep every data which got send through it.

I'd like very much to be able to view what the original message was, and what got changed in it (like Facebook does)
Would be best to make a permission for it, if a user can view it or not.

Kind Regards, 
Julian

**Comment by adriweb** [comment]
Agree, would be nice to be able to see the messages' edit history (maybe like diffs between messages revisions)

**Comment by Xerionty** [comment]
Would be nice to have such feature.

**Comment by kolesar-andras** [comment]
+1

**Comment by dusatvoj** [comment]
Any updates here?

**Comment by randomsymbols** [comment]
Need the feature also. Plus one.

**Comment by ryanvito** [comment]
I am also looking for this feature. It seems there is an api to get to "deleted" messages (sidenote, that gives me errors) but I cannot see what a message was changed to. That's a risk if someone sends a harassment message and then edits it. It appears the edit history is gone and the original document in Mongo is modified instead of appended to with a new message. Thank you for your efforts!
